app/models/concerns/with_code.rb

```diff
@@ -0,0 +1,19 @@
+module WithCode
```

Comment:
Since we use the term code also for referring source-code, I'd rename this to `WithUniqueCode` or `WithGeneratedCode` or something like that.

Reply:
Besides of that, kudos for extracting this mixin :muscle: